packages/babel-plugin-transform-classes/test/fixtures/loose/super-class-id-member-expression/output.js

```diff
@@ -15,8 +15,8 @@ var BaseController2 =
 function (_Chaplin$Controller$A) {
   babelHelpers.inheritsLoose(BaseController2, _Chaplin$Controller$A);
 
-  function BaseController2() {
-    return _Chaplin$Controller$A.apply(this, arguments) || this;
+  function BaseController2(...args) {
+    return _Chaplin$Controller$A.call.apply(_Chaplin$Controller$A, [this].concat(args)) || this;
```

Comment:
I wonder how this might affect class instantiation performance? My main concern with this change is that it increases output complexity for an extremely edge-casey issue that doesn't seem like anyone would rely on.

Reply:
I'm working on a total refactor here, it's just really difficult because the code's a bit low quality at the moment. This will be optimized into a straight `#apply` when I'm done.

Reply:
See https://github.com/babel/babel/pull/7678#discussion_r179643407; `helper-optimise-call-expression` just doesn't deal with this corner case and is inserting a `.call` instead of an `.apply`